cmd/packageloader/loader/packageList.go

```diff
@@ -0,0 +1,48 @@
+package loader
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+	"fmt"
+
+	"github.com/skycoin/cx/cmd/packageloader/bolt"
+	"github.com/skycoin/cx/cmd/packageloader/redis"
+	"golang.org/x/crypto/blake2b"
+)
+
+type PackageList struct {
+	Packages []string
+}
+
+func (pl PackageList) MarshalBinary() ([]byte, error) {
+	return json.Marshal(pl)
+}
+
+func (pl *PackageList) UnmarshalBinary(data []byte) error {
+	return json.Unmarshal(data, pl)
+}
+
+// Encode a package and put it in the specified package list
+func (packageList *PackageList) hashPackage(newPackage *Package) error {
+	var buffer bytes.Buffer
+	encoder := gob.NewEncoder(&buffer)
+
+	err := encoder.Encode(newPackage)
+	if err != nil {
+		return err
+	}
+	h := blake2b.Sum512(buffer.Bytes())
+	packageList.Packages = append(packageList.Packages, fmt.Sprintf("%x", h[:]))
+	switch DATABASE {
+	case "redis":
+		redis.Add(fmt.Sprintf("%x", h[:]), *newPackage)
+	case "bolt":
+		value, err := newPackage.MarshalBinary()
+		if err != nil {
+			return err
+		}
+		bolt.Add(fmt.Sprintf("%x", h[:]), value)
+	}
+	return nil
+}
```

Comment:
Can you separate this one? Just make a function for hashing and will return the hash. The hash function should not be a method of packageList. 

Then for packageList you can add a method that appends a package to it.

Then you can also have a function like SaveToDatabase() that contains the switch statement whether to save to redis or bolt